microsoft/testsuites/security/fips.py

```diff
@@ -19,9 +362,111 @@
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any]
+    ) -> None:
+        # Skip the test if the image is not FIPS enabled.
+        ensure_fips_expectations(log, node, variables, should_be_fips=True)
+
+        # Ensure the system is FIPS enabled.
+        azl_fips = AzlFips.create_instance(log, node)
+        azl_fips.assert_fips_enabled()
+
+        log.info("FIPS is enabled and working correctly.")
+
+    @TestCaseMetadata(
+        description="""
+            Ensures that an AZL machine is not FIPS enabled.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_os=[CBLMariner],
+        ),
+    )
+    def verify_fips_is_disabled_azl(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any]
+    ) -> None:
+        # Skip the test if the image is FIPS enabled.
+        ensure_fips_expectations(log, node, variables, should_be_fips=False)
+
+        # Ensure the system is not FIPS enabled.
+        azl_fips = AzlFips.create_instance(log, node)
+        azl_fips.assert_fips_disabled()
+
+        log.info("FIPS is disabled and properly.")
+
+    @TestCaseMetadata(
+        description="""
+            This test case will
+            1. Enable FIPS on the AZL machine
+            2. Restart the machine
+            3. Verify that FIPS was enabled properly
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[CBLMariner],
+        ),
+    )
+    def verify_fips_enable_azl(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any]
+    ) -> None:
+        # Skip the test if the image is already FIPS enabled.
+        ensure_fips_expectations(log, node, variables, should_be_fips=False)
+
+        # Enable FIPS on the system and make sure it is worked.
+        azl_fips = AzlFips.create_instance(log, node)
+        azl_fips.enable_fips()
+        node.reboot()
+        azl_fips.assert_fips_enabled()
+
+        log.info("Successfully enabled FIPS.")
+
+    @TestCaseMetadata(
+        description="""
+            This test case will
+            1. Disable FIPS on the AZL machine
+            2. Restart the machine
+            3. Verify that FIPS is disabled
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[CBLMariner],
+        ),
+    )
+    def verify_fips_disable_azl(
+        self,
+        log: Logger,
+        node: Node,
+        variables: Dict[str, Any]
+    ) -> None:
+        # Skip the test if the image is already not FIPS enabled.
+        ensure_fips_expectations(log, node, variables, should_be_fips=True)
+
+        # Disable FIPS on the system and make sure it is worked.
+        azl_fips = AzlFips.create_instance(log, node)
+        azl_fips.disable_fips()
+        node.reboot()
+        azl_fips.assert_fips_disabled()
+
+        log.debug("Successfully disabled FIPS.")
```

Comment:
It can be info level to output a successful information.

Reply:
Restore it to its original status, since the environment may be reused.